backend/src/plugin/plugin_core.hpp

```diff
@@ -85,53 +97,193 @@ class Plugin_manager_base
     static_assert(std::is_base_of_v<Plugin_base, Plugin>,
                   "Plugin must be derived from Plugin_base");
 
+protected:
+    explicit Plugin_manager_base(std::set<std::filesystem::path> default_paths)
+        : _default_plugin_paths(std::move(default_paths))
+    {
+    }
+
+    // TODO: figure out how to ignore cognitive complexity warnings induced by logging macros
+    // NOLINTBEGIN(readability-function-cognitive-complexity)
+    std::set<std::filesystem::path> resolve_default_paths() const
+    {
+        std::filesystem::path base_dir;
+        try
+        {
+            base_dir = plugin::Shared_library::get_current_module_directory();
+        }
+        catch(const Hipdnn_exception& e)
+        {
+            HIPDNN_LOG_WARN(
+                "Failed to resolve module directory, will use unresolved default paths: {}",
+                e.get_message());
+            // Fallback to using original, unresolved paths. TODO: possibly remove.
+            return _default_plugin_paths;
+        }
+
+        std::set<std::filesystem::path> resolved_paths;
+
+        for(const auto& path : _default_plugin_paths)
+        {
+            if(path.is_relative())
+            {
+                resolved_paths.insert(base_dir / path);
+            }
+            else
+            {
+                resolved_paths.insert(path);
+            }
```

Comment:
alternative:
```suggestion
            resolved_paths.insert(path.is_relative() ? base_dir / path : path);
```
Also, is path guaranteed to be relative from base_dir?